### PR TITLE
 ridehail warnings, string interpolation and cleanup fixed #620

### LINKED ISSUE #620: fix: ridehail warnings and code cleanup

### PR DESCRIPTION
**Pull request by zishanbilal**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/621)

### CODE REVIEW

File: src/main/scala/beam/agentsim/agents/ridehail/OutOfServiceVehicleManager.scala [hunk 1]
```diff
@@ -49,13 +41,13 @@ class OutOfServiceVehicleManager(
       )
   }
 
-  def registerTrigger(vehicleId: Id[Vehicle], triggerId: Option[Long]) = {
+  def registerTrigger(vehicleId: Id[Vehicle], triggerId: Option[Long]): Option[Option[Long]] = {
```
Comment (kirillm-nor) [inline src/main/scala/beam/agentsim/agents/ridehail/OutOfServiceVehicleManager.scala]: Option[Option[Long]] I think one level optional is enough, lets flatten it

File: src/main/scala/beam/agentsim/agents/vehicles/BicycleFactory.scala [hunk 2]
```diff
@@ -1,8 +1,9 @@
 package beam.agentsim.agents.vehicles
+
 import beam.agentsim.agents.vehicles.EnergyEconomyAttributes.Powertrain
 import beam.sim.BeamServices
-import org.matsim.api.core.v01.{Id, Scenario}
 import org.matsim.api.core.v01.population.Person
+import org.matsim.api.core.v01.{Id, Scenario}
 import org.matsim.households.Household
 
 import scala.collection.JavaConverters
```
Comment (kirillm-nor) [inline src/main/scala/beam/agentsim/agents/vehicles/BicycleFactory.scala]: why not to use 
import scala.collection.JavaConverters._
and after _.asScala

File: src/main/scala/beam/agentsim/agents/vehicles/Trajectory.scala [hunk 3]
```diff
@@ -97,4 +90,10 @@ class Trajectory(val path: Vector[SpaceTime]) {
       })
       .sum
   }
+
+  protected[agentsim] def append(newTrajectory: Trajectory): Unit = {
+    this.synchronized {
+      _path = _path ++ newTrajectory._path
```
Comment (kirillm-nor) [inline src/main/scala/beam/agentsim/agents/vehicles/Trajectory.scala]: Let's use concurrent collection and get rid of synchronise block

**Comment by kirillm-nor** [comment]
Just to check, did you run scalafmt before commit?

File: src/main/scala/beam/agentsim/agents/ridehail/OutOfServiceVehicleManager.scala (hunk 1 above)
Comment (zishanbilal) [reply src/main/scala/beam/agentsim/agents/ridehail/OutOfServiceVehicleManager.scala]: the only statement of the method is` triggerIds.put(vehicleId, triggerId)` (put Option[Long] to map) and it returns Option[VAL] that makes it Option[Option[Long]]. Please suggest the fix.

**Comment by zishanbilal** [comment]
> Just to check, did you run scalafmt before commit?

run scalafmt and other fix code against other review comments

File: src/main/scala/beam/agentsim/agents/ridehail/OutOfServiceVehicleManager.scala (hunk 1 above)
Comment (JustinPihony) [reply src/main/scala/beam/agentsim/agents/ridehail/OutOfServiceVehicleManager.scala]: `triggerIds.put(vehicleId, triggerId).flatten`

**Pull request closed without merging**